### PR TITLE
[ENH] golden output tests for `FlowStateForecaster`

### PR DESCRIPTION
**Pull request by AbdulAliMamnun**
Parts of this PR were drafted with Claude Code — the new test function and its reference-case table. The estimator selection, the reference-output generation, and the verification below are my own; both reference triples were measured by me from the upstream implementation.

#### Reference Issues/PRs

Part of #10672. Follows #10627 / #10674, and the pattern I used in #11043.

#### What does this implement/fix? Explain your changes.

Adds a golden-output regression test for `FlowStateForecaster`, parametrized over the default configuration and `scale_factor=0.5`.

`FlowStateForecaster` is a thin wrapper over `tsfm_public.FlowStateForPrediction`: `_run` reshapes the context to `(1, n, 1)`, casts to the model's dtype and device, makes one forward call, and takes `prediction_outputs[0]`. This test pins that plumbing — context shaping, `pred_len` derivation from `max(fh)`, and output filtering — against the source implementation called directly.

References were generated with this script, also embedded as a comment in the test file:

```python
import numpy as np, torch
from tsfm_public import FlowStateForPrediction

AIRLINE = [...]                      # sktime's load_airline values
y_train, pred_len = AIRLINE[:-12], 12

model = FlowStateForPrediction.from_pretrained(
    "ibm-research/flowstate", revision="r1.1", batch_first=True
).to("cpu")
model.eval()

past = torch.tensor(
    np.asarray(y_train, dtype=np.float32).reshape(1, -1, 1),
    dtype=model.dtype, device=model.device,
)
with torch.inference_mode():
    out = model(
        past_values=past, prediction_length=pred_len, batch_first=True,
        scale_factor=1.0, prediction_type="mean",
    )
print(out.prediction_outputs.detach().cpu().numpy()[0].ravel()[:3])
# -> [399.65942, 389.4466, 432.6386]
```

`scale_factor=0.5` differs only in that argument, giving `[426.69104, 429.21277, 460.16183]`. The two cases differ by 27–40 in absolute terms, so the second genuinely exercises a different path rather than duplicating the first.

`revision="r1.1"` is forwarded to `from_pretrained` by the wrapper, so the checkpoint is version-pinned. It resolved to commit `940580f969036cfca514cdff725f7b02c03e1bcf`, recorded in a comment.

FlowState does no sampling — `_run` is a single forward pass under `torch.inference_mode()` after `.eval()`, and quantiles come off `quantile_outputs` rather than sampled paths — so there is no seed to control and reruns reproduce bit-exactly across separate processes.

#### Does your contribution introduce a new dependency? If yes, which one?

No.

#### What should a reviewer concentrate their feedback on?

- The diff is purely additive: 56 insertions, no deletions. `test_flowstate.py` already contained `test_flowstate_predict_proba_histogram_consistency`; its body, its `_check_estimator_deps` gate, and the module docstring are unchanged. The new test carries its own per-test `run_test_for_class` skipif rather than a module-level `pytestmark`, so the existing test's run conditions are untouched.
- Tolerances follow `test_ttm.py` (`rtol=1e-5, atol=1e-4`). References were generated on CPU; locally sktime resolves to MPS, where the third element drifts by 2.2e-4 — past `atol` on its own, absorbed by the `rtol` term at these magnitudes. `sktime/tests/_test_vm.py` disables MPS on Darwin runners, so CI compares CPU against CPU and the values match exactly.
- I did not pin the device in the test. `FlowStateForecaster` has no `device` parameter — `_resolve_device()` picks cuda → mps → cpu implicitly — so pinning would mean reaching past the public API.

#### Did you add any tests for the change?

Yes, this PR is the test. `tests:specific` on `FlowStateForecaster` updated to `["sktime.forecasting.tests.test_flowstate"]`, verified to resolve through `_get_estimator_specific_test_modules`.

- Both cases verified against the upstream references.
- Upstream reference reruns are bit-identical across separate processes.
- Mutation check: perturbing the `scale_factor=1.0` reference by 10.0 fails at index 0 with max absolute difference 9.999939, while the `scale_factor=0.5` case passes unaffected — so the assertion reaches the reference values and the cases are isolated. Reverted and confirmed byte-identical to the commit.
- `pre-commit` clean on both files.

`FlowStateForecaster` carries `tests:vm: True`, so the golden test skips under plain local pytest and executes in the VM CI job — same as `test_ttm.py`.

#### Any other comments?

@geetu040 — noting this touches an estimator migrated in #10508. The test asserts on outputs rather than internals, so it should survive the move to `BaseFoundationForecaster` if the numerics are preserved, and flag it if they aren't. Happy to rebase or adjust if the migration changes the constructor surface.

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/sktime/blob/main/.all-contributorsrc) in the `sktime` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://github.com/all-contributors/allcontributors.org/blob/main/src/content/docs/en/reference/emoji-key.mdx)
- [ ] Optionally, for added estimators: I've added myself and possibly to the `maintainers` tag - do this if you want to become the owner or maintainer of an estimator you added.
  See here for further details on the [algorithm maintainer role](https://www.sktime.net/en/latest/get_involved/governance.html#algorithm-maintainers).
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

### CODE REVIEW

**Review by geetu040 — APPROVED** [review]
Thanks, changes look good, waiting for CI.
